app/forms/form.rb

```diff
@@ -35,9 +37,13 @@ def i18n_errors_path(msg)
     I18n.t("#{i18n_namespace}.forms.#{i18n_form_namespace}.errors.#{msg}")
   end
 
+  def permitted_params
+    @permitted_params ||= params.fetch(:claim, {}).permit(*attributes)
```

Comment:
thoughts on changing the initialiser to something like
```
  def initialize(claim:, journey:, params:)
    @claim = claim
    @journey = journey
    super(permitted_params)
  end
```
then the subclasses don't have to do the `self.some_attribute = permitted_params[:some_attribute]` ?

Reply:
@rjlynch good idea! this seems to do the trick:

```ruby
  def initialize(claim:, journey:, params:)
    super
    assign_attributes(permitted_params)
  end
```

Reply:
Actually thinking about it we might not be able to do this.
We probably want to support defaulting to any attributes on the claim / eligibility if they're not set by the params so we can support remembering the user's options when they navigate back. If we were just working with a Claim we could do something like
`assign_attributes(permitted_params.reverse_merge(claim.attributes.slice(attributes))` but as we're potentially dealing with Eligibility defaulting the params might get ugly. Maybe best to leave this for the minute and let subclasses initialise their own attributes 🤔

Reply:
Also in the `PoorPerformanceForm` I think we probably need to set the default values from the eligibility record sort of like

```ruby
def initialize(journey:, claim:, params:)
  self.subject_to_formal_performance_action = permitted_params.fetch(:subject_to_formal_performance_action, claim.eligibility.subject_to_formal_performance_action)
 self.subject_to_disciplinary_action = permitted_params.fetch(:subject_to_disciplinary_action, claim.eligibility.subject_to_disciplinary_action)
end
```
so navigating back doesn't lose the previous option (if we care about that!)

Reply:
I see what you are saying; you are right that it loses the values; they should be pre-filled from the model if they exist. I see that for the first form object (current school), we did a bit of a hack, as there is an instance method on the form object exposing a certain attribute from the claim/eligibility, so I completely missed the defaulting.

Reply:
> as we're potentially dealing with Eligibility defaulting the params might get ugly. Maybe best to leave this for the minute and let subclasses initialise their own attributes until a pattern emerges

definitely my thinking, also I know we actually want to remove that model soon.

Reply:
Video as I'm not sure if I'm being clear 😂 

Without setting attribute defaults from eligibility record
![without-default-attributes](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/assets/9936028/92ad89bd-dcf2-4883-ae15-97596b0b5c87)



With setting attribute defaults from eligibility record
![with-default-attributes](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/assets/9936028/4ce4fcc2-f673-4266-a307-9aac5c48caa5)

Reply:
Got it, you were clear thanks. I'll fix that

Reply:
@rjlynch I agree with Steve here: https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2664/files#r1553830035

> [...] Perhaps a convenience method in the base class to read previous answers would be helpful and allow us to swap out the underlying data source more easily.

I'll try to look at this first thing next week, unless you come up with something like that before... the eligibility model is a bit of a problem to deal with but 🤷

Reply:
I think you're right, but bear in mind we want to get rid of the `Claim` record in the next phase so will want to pull in the previously filled answers from the replacement data source (possibly just a Hash in the session), so we could have a convenience method to access the existing answers in the same way we have an `update!` method in the base `Form` class to abstract access to this.

Reply:
@rjlynch I've reverted the changes that I had introduced as part of this conversation so we can tackle it properly separately but made sure we get the default values from the eligibility model where needed